src/sectionproperties/analysis/fea.py

```diff
@@ -845,23 +856,18 @@ def gauss_points(n: int) -> np.ndarray:
             ],
             dtype=float,
         )
-    else:
-        raise ValueError("n must be 1, 3 or 6.")
 
+    raise ValueError("n must be 1, 3, 4 or 6.")
 
-def shape_function(
-    coords: np.ndarray,
-    gauss_point: tuple[float, float, float, float],
-) -> tuple[np.ndarray, np.ndarray, float]:
-    """Calculates shape functions, derivates and the Jacobian determinant.
 
-    Computes the shape functions, shape function derivatives and the determinant of the
-    Jacobian matrix for a ``Tri6`` element at a given Gauss point.
+@lru_cache(maxsize=None)
+def __shape_function_cached(coords: tuple, gauss_point: tuple):
```

Comment:
I'm trying to make `sectionproperties` as "strictly typed" as possible - suggest:

```python
def __shape_function_cached(
    coords: tuple[float, ...],
    gauss_point: tuple[float, float, float],
) -> tuple[np.ndarray, np.ndarray, float]:
```